.gitlab-ci.yml

```diff
@@ -55,3 +55,21 @@ build:publish-metadata-writer:
         .
     - docker login --username ${DOCKER_USERNAME} --password ${DOCKER_API_KEY} ${DOCKER_REPO_URL}
     - docker push ${IMAGE_REPOSITORY_TAG}
+
+build:publish-front-end:
+  extends: .aip_python_debian_image
+  stage: build
+  variables:
+    ARTIFACT_NAME: 'zillow-kfp-frontend'
+  before_script:
+    - !reference [.generate_docker_image_version, before_script]
+  script:
+    - |
+      docker build \
+        --tag ${IMAGE_REPOSITORY_TAG} \
+        --build-arg COMMIT_HASH=${COMMIT_HASH} \
+        --build-arg DATE="`date -u`" \
+        --file frontend/Dockerfile \
+        .
+    - docker login --username ${DOCKER_USERNAME} --password ${DOCKER_API_KEY} ${DOCKER_REPO_URL}
+    - docker push ${IMAGE_REPOSITORY_TAG}
```

Comment:
Shall we worry about image tag collision with the `build:publish-metadata-writer` step above? Looks like both gitlab ci step runs `docker push ${IMAGE_REPOSITORY_TAG}`

Reply:
It's based off of https://github.com/zillow/pipelines/blob/9fbe629a86ca3bef907b81400463418cb0b3f5b4/.gitlab-ci.yml#L63 `ARTIFACT_NAME` see [code link](https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/aip-infrastructure/ci-templates/ci-cd-template/-/blob/main/blocks/docker.yml#L35-40)

also see how the image names are different in the following pipeline:
 https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/sdks/internal/zillow-kfp/-/pipelines/2884129

Reply:
Understood. Thanks for pointing that out.